lib/rules/valid-typeof.js

```diff
@@ -37,8 +47,12 @@ module.exports = {
                     if (parent.type === "BinaryExpression" && OPERATORS.indexOf(parent.operator) !== -1) {
                         const sibling = parent.left === node ? parent.right : parent.left;
 
-                        if (sibling.type === "Literal" && VALID_TYPES.indexOf(sibling.value) === -1) {
-                            context.report(sibling, "Invalid typeof comparison value.");
+                        if (sibling.type === "Literal") {
+                            if (VALID_TYPES.indexOf(sibling.value) === -1) {
+                                context.report(sibling, "Invalid typeof comparison value.");
+                            }
+                        } else if (context.options[0] && context.options[0].requireStringLiterals) {
+                            context.report(sibling, "Invalid typeof comparison to a non-string literal.");
```

Comment:
To avoid the double negative, what about saying "Typeof comparisons should be to string literals."? (I'm not thrilled with the wording of that either, so I'd be open to better phrasings.)